airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -211,6 +214,15 @@ def _evaluate_trigger_rule(
                     "upstream_tasks_state={2}, upstream_task_ids={3}"
                     .format(tr, num_failures, upstream_tasks_state,
                             task.upstream_task_ids))
+        elif tr == TR.NONE_FAILED_OR_SKIPPED:
+            num_failures = upstream - successes - skipped
```

Comment:
So actually the name should be `all_success_or_skipped` would be more accurate?

Reply:
I'm not entirely sure. I'm kind of confused how it differs from NONE_FAILED, but no one seems to have an issue with it upstream :/

Reply:
Yeah the implementation looks exactly the same :/

Reply:
Actually no. For `NONE_FAILED_OR_SKIPPED`, the task won't be skipped if there is at least one success